jaxlib/gpu/sparse.cc

```diff
@@ -565,32 +565,37 @@ size_t Gtsv2BufferSize(F f, int m, int n, int ldb) {
 }
 
 size_t Gtsv2BufferSizeF32(int m, int n, int ldb) {
-  return Gtsv2BufferSize(hipsparseSgtsv2_bufferSizeExt, m, n, ldb);
+  return Gtsv2BufferSize(gpusparseSgtsv2_bufferSizeExt, m, n, ldb);
 }
 
 size_t Gtsv2BufferSizeF64(int m, int n, int ldb) {
-  return Gtsv2BufferSize(hipsparseDgtsv2_bufferSizeExt, m, n, ldb);
+  return Gtsv2BufferSize(gpusparseDgtsv2_bufferSizeExt, m, n, ldb);
 }
 
 py::dict Registrations() {
   py::dict dict;
-  dict["hipsparse_csr_todense"] = EncapsulateFunction(CsrToDense);
-  dict["hipsparse_csr_fromdense"] = EncapsulateFunction(CsrFromDense);
-  dict["hipsparse_csr_matvec"] = EncapsulateFunction(CsrMatvec);
-  dict["hipsparse_csr_matmat"] = EncapsulateFunction(CsrMatmat);
-  dict["hipsparse_coo_todense"] = EncapsulateFunction(CooToDense);
-  dict["hipsparse_coo_fromdense"] = EncapsulateFunction(CooFromDense);
-  dict["hipsparse_coo_matvec"] = EncapsulateFunction(CooMatvec);
-  dict["hipsparse_coo_matmat"] = EncapsulateFunction(CooMatmat);
-  dict["hipsparse_gtsv2_f32"] = EncapsulateFunction(gtsv2_f32);
-  dict["hipsparse_gtsv2_f64"] = EncapsulateFunction(gtsv2_f64);
+#if JAX_GPU_HAVE_SPARSE
+  dict[JAX_GPU_PREFIX "sparse_csr_todense"] = EncapsulateFunction(CsrToDense);
+  dict[JAX_GPU_PREFIX "sparse_csr_fromdense"] =
+      EncapsulateFunction(CsrFromDense);
+  dict[JAX_GPU_PREFIX "sparse_csr_matvec"] = EncapsulateFunction(CsrMatvec);
+  dict[JAX_GPU_PREFIX "sparse_csr_matmat"] = EncapsulateFunction(CsrMatmat);
+  dict[JAX_GPU_PREFIX "sparse_coo_todense"] = EncapsulateFunction(CooToDense);
+  dict[JAX_GPU_PREFIX "sparse_coo_fromdense"] =
+      EncapsulateFunction(CooFromDense);
+  dict[JAX_GPU_PREFIX "sparse_coo_matvec"] = EncapsulateFunction(CooMatvec);
+  dict[JAX_GPU_PREFIX "sparse_coo_matmat"] = EncapsulateFunction(CooMatmat);
+#endif
+  dict[JAX_GPU_PREFIX "sparse_gtsv2_f32"] = EncapsulateFunction(gtsv2_f32);
+  dict[JAX_GPU_PREFIX "sparse_gtsv2_f64"] = EncapsulateFunction(gtsv2_f64);
   // TODO(tomhennigan): Add support for gtsv2 complex 32/64.
   return dict;
 }
 
-PYBIND11_MODULE(_hipsparse, m) {
-  m.attr("hipsparse_supported") = py::bool_(true);
+PYBIND11_MODULE(_sparse, m) {
+  m.attr("cusparse_supported") = py::bool_(JAX_GPU_HAVE_SPARSE);
```

Comment:
Add m.attr("hipsparse_supported") = py::bool_(JAX_GPU_HAVE_SPARSE); ?

Reply:
I changed this to `sparse_supported` (removed the prefix) and updated the Python code to match.

Reply:
Ah yeah - that's better.  Just retested with this change and it worked.